enterprise/server/scheduling/scheduler_server/scheduler_server.go

```diff
@@ -1361,10 +1361,12 @@ func (s *SchedulerServer) enqueueTaskReservations(ctx context.Context, enqueueRe
 				// (in subsequent loop iterations) if the preferred node probe fails.
 				preferredNode = nil
 			} else {
+				nodeBalancer.mu.Lock()
```

Comment:
I think it would be better to add methods to nodePool to retrieve the nodes under lock rather than accessing the mutex directly.

Reply:
Done, wdyt?

Reply:
looks good!

Reply:
thanks for fixing this